### PR TITLE
More NativeAOT annotations

### LINKED ISSUE #4387: NativeAOT + Asp.Net Core = Missing constructor accepting 'DataOptions'
### Describe your issue

I'm trying to use the Linq2db in Native AOT project and facing an unexpected issue.
In the dev environment  "on my machine" everything works fine, but when I'm trying to build a docker image with published AOT app and run it, I'm seeing the following exception.

```
Unhandled Exception: System.ArgumentException: Missing constructor accepting 'DataOptions' on type ExpenseDb.
   at LinqToDB.AspNet.ServiceConfigurationExtensions.HasTypedContextConstructor[TContext]() + 0x135
   at LinqToDB.AspNet.ServiceConfigurationExtensions.AddLinqToDBContext[TContext,TContextImplementation](IServiceCollection, Func`3, ServiceLifetime) + 0x45
   at Program.<Main>$(String[] args) + 0x271
   at app!<BaseAddress>+0x135456c
```

### Steps to reproduce

The DataConnection class looks pretty standard according to the documentation:

```c#
// DataConnection
//
public class ExpenseDb(DataOptions<ExpenseDb> options) : DataConnection(options.Options)
{
   //...
}

// Registration
//
// Add Database context
//
var connectionString = builder.Configuration.GetConnectionString("Default");
builder.Services.AddLinqToDBContext<ExpenseDb>((provider, options)
    => options
        .UsePostgreSQL(connectionString!, PostgreSQLVersion.v15)
        .UseDefaultLogging(provider));

```

### Environment details

`Linq To DB` version: *5.3.2*

Database (with version): *Postgres 15* 

ADO.NET Provider (with version): *Npgsql 8.0.1* 

Operating system: *alpine-3.18*

.NET Version: *8.0.1*

### PR DESCRIPTION
**Pull request by MaceWindu**
Fix #4387

### CODE REVIEW

File: Source/LinqToDB.AspNet/ServiceConfigurationExtensions.cs [hunk 1]
```diff
@@ -97,8 +102,12 @@ public static class ServiceConfigurationExtensions
 		/// <returns>
 		///     The same service collection so that multiple calls can be chained.
 		/// </returns>
-		public static IServiceCollection AddLinqToDBContext<TContext>(
-			this IServiceCollection serviceCollection,
+		public static IServiceCollection AddLinqToDBContext<
+#if NET6_0_OR_GREATER
+			[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)]
```
Comment (viceroypenguin) [inline Source/LinqToDB.AspNet/ServiceConfigurationExtensions.cs:107]: Is this not added via PolySharp?

**Review by viceroypenguin — DISMISSED** [review]

File: Source/LinqToDB.AspNet/ServiceConfigurationExtensions.cs (hunk 1 above)
Comment (MaceWindu) [reply Source/LinqToDB.AspNet/ServiceConfigurationExtensions.cs:107]: hmm, should be. Maybe we need to update it

**Comment by MaceWindu** [comment]
I will perform couple of simple tests to see if there are other issues

**Comment by MaceWindu** [comment]
doesn't work, and looking at error message that is not helpfull at all I think I don't care much about native aot for now....


```
System.NotSupportedException: 'System.Func`2[LinqToDB.DataProvider.PostgreSQL.NpgsqlProviderAdapter+NpgsqlDbType,NpgsqlTypes.NpgsqlDbType]' is missing native code or metadata. This can happen for code that is not compatible with trimming or AOT. Inspect and fix trimming and AOT related warnings that were generated when the app was published. For more information see https://aka.ms/nativeaot-compatibility
```

rd.xml doesn't help

```xml
<Directives xmlns="http://schemas.microsoft.com/netfx/2013/01/metadata">
	<Application>
		<Assembly Name="linq2db" Dynamic="Required All" />
		<Assembly Name="Npgsql" Dynamic="Required All" />
	</Application>
</Directives>

```

**Comment by MaceWindu** [comment]
Maybe later we will find some hundreds hours to annotate it all, but for now users should play with rd.xml. Maybe they will have more luck than me

**Pull request merged**